cli/command/container/run.go

```diff
@@ -59,6 +61,9 @@ func NewRunCommand(dockerCli *command.DockerCli) *cobra.Command {
 	flags.StringVar(&opts.name, "name", "", "Assign a name to the container")
 	flags.StringVar(&opts.detachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
 
+	flags.StringVar(&opts.platform, "platform", "", "Set platform if server is multi-platform capable")
+	flags.SetAnnotation("platform", "version", []string{"1.32"})
```

Comment:
Since this is going to be repeated in a bunch of commands how about we move this to 

```go
func addPlatformFlag(flags *pflag.FlagSet, opt *PlatformOpt)
```

in `cli/flags/common.go`


There will also need to be another annotation for hiding based on the header:

```go
flags.SetAnnotation("platform", "platform", nil)
```

The code to hide flags based on annotations is in `cmd/docker/docker.go`